src/lightning_fabric/loggers/tensorboard.py

```diff
@@ -0,0 +1,310 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import os
+from argparse import Namespace
+from typing import Any, Dict, Mapping, Optional, TYPE_CHECKING, Union
+
+import numpy as np
+from lightning_utilities.core.imports import RequirementCache
+from torch import Tensor
+from torch.nn import Module
+
+from lightning_fabric.loggers.logger import Logger, rank_zero_experiment
+from lightning_fabric.utilities.cloud_io import get_filesystem
+from lightning_fabric.utilities.logger import _add_prefix, _convert_params, _flatten_dict
+from lightning_fabric.utilities.logger import _sanitize_params as _utils_sanitize_params
+from lightning_fabric.utilities.rank_zero import rank_zero_only, rank_zero_warn
+from lightning_fabric.utilities.types import _PATH
+
+log = logging.getLogger(__name__)
+
+_TENSORBOARD_AVAILABLE = RequirementCache("tensorboard")
+_TENSORBOARDX_AVAILABLE = RequirementCache("tensorboardX")
+if TYPE_CHECKING:
+    # assumes at least one will be installed when type checking
+    if _TENSORBOARD_AVAILABLE:
+        from torch.utils.tensorboard import SummaryWriter
+    else:
+        from tensorboardX import SummaryWriter  # type: ignore[no-redef]
+
+
+class TensorBoardLogger(Logger):
```

Comment:
Don't we want the `pytorch_lightning.loggers.tensorboard` module to import this class? Without deprecations

Reply:
I think we want to eventually. In Lite however we currently have more freedom to make minor changes without triggering us to do immediately do refactors that touch many areas of Trainer.

Reply:
Yes, let's make them as we have the freedom to do so.

Reply:
I don't think we are planning any changes to the logger objects themselves. Do you have any in mind? And any differences could be resolved by inheritance if necessary later.

Reply:
The changes are listed in the PR summary.
Also this: https://github.com/Lightning-AI/lightning/pull/16121#discussion_r1054375539